### PR TITLE
Fix a bug in BatchNorm

### PR DESCRIPTION
**Pull request by kakao-steve-ai**
## 🙏 Describe the pull request

FP16 타입에 대한 Backward 오류 수정했습니다.

## ✅ Checklist

- [x] Code follows the project's coding conventions and style.
- [x] Tests have been added or updated to cover the changes.
- [x] Documentation has been updated, if necessary.

### CODE REVIEW

File: trident/kernel/batch_norm.py [hunk 1]
```diff
@@ -152,7 +153,7 @@ def backward(
         grad_mean = tl.sum(tl.where(condition, grad_centered_mean, 0.0) / denominator)
         grad_input = grad_centered_mean - grad_mean
 
-        tl.store(grad_input_block_ptr, grad_input, boundary_check=(0, 1))
+        tl.store(grad_input_block_ptr, grad_input.to(dtype), boundary_check=(0, 1))
```
Comment (mejai1206) [inline trident/kernel/batch_norm.py:156]: `grad_weight`는 캐스팅이 필요없는걸까요??

Comment (kakao-steve-ai) [reply trident/kernel/batch_norm.py:156]: `grad_weight`는 캐스팅을 안해도 에러가 안나내요. 추측하기로는, 나누기 `/` 나 `tl.sqrt`는 upcasting을 하지 않는것 같습니다. `grad_input`은 언제 어떻게 upcasting 된건지 좀 알아볼게요.

**Review by mejai1206 — APPROVED** [review]
LGTM

**Review by daemyung — APPROVED** [review]
LGTM

File: trident/kernel/batch_norm.py (hunk 1 above)
Comment (daemyung) [reply trident/kernel/batch_norm.py:156]: `tl.where`에서 업케스팅이 됩니다. https://github.com/openai/triton/issues/1891 참고요.

Comment (kakao-steve-ai) [reply trident/kernel/batch_norm.py:156]: @daemyung 좋은 정보 감사합니다.

Comment (kakao-steve-ai) [reply trident/kernel/batch_norm.py:156]: 그러면 `grad_weight`도 캐스팅이 필요하겠네요.(`centered_mean`구할때 `where`를 사용해서 이미 업케스팅이 되었을거라서요.) 
`grad_weight_ptr`이 블록포인터를 안 사용해서 컴파일 타임에 에러가 안났던거 같습니다.
